jlm/llvm/opt/alias-analyses/MemoryStateEncoder.cpp

```diff
@@ -494,81 +423,96 @@ class RegionalizedStateMap final
     return GetStateMap(region).GetState(memoryNode);
   }
 
+  std::vector<StateMap::MemoryNodeStatePair *>
+  GetStates(
+      const rvsdg::Region & region,
+      const util::HashSet<const PointsToGraph::MemoryNode *> & memoryNodes)
+  {
+    return GetStateMap(region).GetStates(memoryNodes);
+  }
+
+  std::vector<StateMap::MemoryNodeStatePair *>
+  GetStates(const rvsdg::SimpleNode & node) noexcept
+  {
+    return GetStates(*node.region(), GetSimpleNodeModRef(node));
+  }
+
   util::HashSet<const PointsToGraph::MemoryNode *>
-  GetMemoryNodes(const rvsdg::Output & output)
+  GetSimpleNodeModRef(const rvsdg::SimpleNode & node)
   {
-    auto & memoryNodeCache = GetMemoryNodeCache(*output.region());
-    return memoryNodeCache.GetMemoryNodes(output);
+    // TODO: This implementation is temporary, until the new ModRefSummary interface is added
+    util::HashSet<const PointsToGraph::MemoryNode *> result;
+    const auto & addOutputToModRefSet = [&](const rvsdg::Output & output)
+    {
+      auto & trueOutput = LookupAddress(output);
+      auto memoryNodes = ModRefSummary_.GetOutputNodes(trueOutput);
+      result.UnionWithAndClear(memoryNodes);
+    };
+
+    if (jlm::rvsdg::is<StoreOperation>(&node))
+    {
+      addOutputToModRefSet(*StoreOperation::AddressInput(node).origin());
+    }
+    else if (jlm::rvsdg::is<LoadOperation>(&node))
+    {
+      addOutputToModRefSet(*LoadOperation::AddressInput(node).origin());
+    }
+    else if (jlm::rvsdg::is<FreeOperation>(&node))
+    {
+      addOutputToModRefSet(*node.input(0)->origin());
+    }
+    else if (jlm::rvsdg::is<MemCpyOperation>(&node))
+    {
+      addOutputToModRefSet(*node.input(0)->origin());
+      addOutputToModRefSet(*node.input(1)->origin());
```

Comment:
Could we add some convenience functions for retrieving the address inputs/ouputs for these nodes as well, similarly to Load/Store operation?

Reply:
Yes sure I can add them to this PR